### DISCUSSION TITLE
Filter kanban by date comparing to current date

### DISCUSSION THREAD

**Discussion by guilhermeoliveira-dev** [marcusolsson/obsidian-projects#835] [Help] [answered]
I am using the projects plugin to keep track of various chores and events that I need to remember (like college projects and exams). I am also using the kanban view and the calendar view to visually keep track of when things have to be done and if I have done them. 

Howhever, I want the kanban view to automatically remove the older notes, like a exam that I took a month ago or something like that, but I don't think there is a way to do this sort of filtering because the filter by date only seems to accept static dates typed in manually, and it certainly would be against the whole principle of the project to keep changing the limit date manually. 

I am currently doing this by changing the status of the notes from "done" to "hidden" and telling the kanban to hide "hidden" statuses. Is there a way to do this using the templater codes? Like maybe something in the lines of "when currentDate = limitDate + 7, change status to hidden", so that the code changes visibility of the note after 7 days of the limitDate being passed, and therefore hides it.

I'm open to suggestions. And maybe this could even turn out to be a new future to be implemented in future updates, it would be awesome to just filter by current date like the "code" I presented above.

**Comment by Acylation** [comment, marked as answer]
I'm leaning to some internal support for dynamic date ranges. For example, `Due is within x days/weeks`.

**Reply by guilhermeoliveira-dev** [reply]
This would be extremely helpful. When I used notion I had a view that specified in the lines of `any exams or important events within the next week` at the top of my main page. It's really helpful for keeping track of pending chores, and probably many other uses.

**Reply by virtu** [reply]
Just came across this when facing the same problem (using color to highlight projects/tasks (1) due this week and (2) over deadline).

Would be really great to have more fine-grained control than `within x days/weeks`; for example `date is x days (or [more|less]) [before|after] today` (e.g. `date is 7 days or less before today` for projects due in a week or `date is 1 day or more after today` for overdue projects). I assume this might complicate things a bit, but I think it'll make the feature much more usable.
